pkg/appgw/ingress_rules.go

```diff
@@ -18,37 +18,53 @@ func (c *appGwConfigBuilder) getListenersFromIngress(ingress *v1beta1.Ingress, e
 	listeners := make(map[listenerIdentifier]listenerAzConfig)
 
 	// if ingress has only backend configured
-	if ingress.Spec.Backend != nil && len(ingress.Spec.Rules) == 0 {
+	if ingress.Spec.Backend != nil && ingress.Spec.TLS == nil && len(ingress.Spec.Rules) == 0 {
```

Comment:
I think we should not exit this function when TLS is not specified. This will miss the case for hostname-extension.

Can we do the following ? 
Remove
`if ingress.Spec.Backend != nil && ingress.Spec.TLS == nil && len(ingress.Spec.Rules) == 0 {`
and `if len(ingress.Spec.Rules) == 0 {`

and replace with a similar logic:
```go
if ingress.Spec.Backend != nil {
  listeners = c.collectListener(ingress, nil, env, policy)
}
```

Reply:
> I think we should not exit this function when TLS is not specified. This will miss the case for hostname-extension.
> 
> Can we do the following ?
> Remove
> `if ingress.Spec.Backend != nil && ingress.Spec.TLS == nil && len(ingress.Spec.Rules) == 0 {`
> and `if len(ingress.Spec.Rules) == 0 {`
> 
> and replace with a similar logic:
> 
> ```go
> if ingress.Spec.Backend != nil {
>   listeners = c.collectListener(ingress, nil, env, policy)
> }
> ```

 `if ingress.Spec.Backend != nil && ingress.Spec.TLS == nil && len(ingress.Spec.Rules) == 0 ` can be removed

Reply:
added one more function test case to cover two ingresses, one with default backend defined, the other not.